client/components/app.jsx

```diff
@@ -59,6 +65,35 @@ class App extends React.Component {
       .catch(err => console.error(err));
   }
 
+  updateGrade(editedGrade) {
+    const gradeIndex = this.state.grades.findIndex(grade => editedGrade.id === grade.id);
+    const gradesCopy = this.state.grades.slice();
+    gradesCopy[gradeIndex] = editedGrade;
+
+    fetch(`/api/grades/${editedGrade.id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(editedGrade)
+    })
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          grades: gradesCopy,
+          view: 'add'
+        }))
+      .catch(err => console.error(err));
+  }
+
+  getGradeToUpdate(id) {
+    const gradeIndex = this.state.grades.findIndex(grade => id === grade.id);
+    const gradesCopy = this.state.grades.slice();
+    const gradeToUpdate = gradesCopy[gradeIndex];
```

Comment:
Since you are not actually modifying the array, you don't need to slice it here.